tests/tox_env/python/test_python_api.py

```diff
@@ -95,6 +95,7 @@ def test_base_python_env_no_conflict(env: str, base_python: list[str], ignore_co
         ("py3", ["py2"], ["py2"]),
         ("py38", ["py39"], ["py39"]),
         ("py38", ["py38", "py39"], ["py39"]),
+        ("py38", ["python3"], ["python3"]),
```

Comment:
This shouldn't conflict, but `py27` and `python3` should 🤔

Reply:
Both should conflict. `python3` on my system == `python3.11.1`. That is decidedly _not_ `python3.8`, which is what I intended. I should be exploding on that combination - which I do (unless the user explicitly said not to, via `ignore_base_python_conflict`)

Reply:
You're miss understanding. `base_python` **is not** an executable match. It is a specification. And the specification states `Python 3`, without resolving it.

Reply:
That may be the case, but providing a *specification* of `python3` (e.g. `base_python = python3`) causes the `python3` *executable* to be used. For example, using a `tox.ini` like this:

```ini
[tox]
minversion = 3.1
envlist = py{37,38,39,310,311}

[testenv]
basepython = python3
```

If I run `tox -e py38`, the `py38` venv is using `python3.11` :

```
❯ source .tox/py38/bin/activate
❯ python --version
Python 3.11.1
```

No warnings. Nothing. That's got to be incorrect behaviour, surely?

Reply:
Indeed, so we have two options:

- for py38 `python3` spec should conflict and hard fail,
- hard fail once we resolve `python3` and see its something else.

I'm inclined to go for option 1.

Reply:
Agree. Option 1 is the way to go. Option 2 will result in different behaviour depending on the environment making tox less deterministic. That's A Bad Thing :tm:

Without this change, the `py38` venv uses whatever `python3` is on my host (`python3.11`). With this change, things hard fail as expected.

```
❯ tox -e py38
py38: failed with env name py38 conflicting with base python python3
  py38: FAIL code 1 (0.00 seconds)
  evaluation failed :( (0.08 seconds)
```

So it seems I have implemented option 1, yes?

Reply:
What happens for env `py3`?

Reply:
```
❯ tox -e py3
py3: install_deps> python -I -m pip install -r /tmp/tox-issue-2717/test-requirements.txt
.pkg: _optional_hooks> python /tmp/tox-venv/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
.pkg: get_requires_for_build_sdist> python /tmp/tox-venv/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
.pkg: prepare_metadata_for_build_wheel> python /tmp/tox-venv/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
.pkg: build_sdist> python /tmp/tox-venv/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
py3: install_package_deps> python -I -m pip install requests
py3: install_package> python -I -m pip install --force-reinstall --no-deps /tmp/tox-issue-2717/.tox/.tmp/package/11/foo-0.0.0.tar.gz
.pkg: _exit> python /tmp/tox-venv/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
  py3: OK (4.39 seconds)
  congratulations :) (4.42 seconds)
```